test/integration/particle_emitter.cc

```diff
@@ -0,0 +1,170 @@
+#include "ignition/gazebo/test_config.hh"
+
+#include "helpers/Relay.hh"
+
+using namespace ignition;
+using namespace gazebo;
+
+class ParticleEmitterTest : public ::testing::Test
+{
+  // Documentation inherited
+  protected: void SetUp() override
+  {
+    ignition::common::Console::SetVerbosity(4);
+    setenv("IGN_GAZEBO_SYSTEM_PLUGIN_PATH",
+           (std::string(PROJECT_BINARY_PATH) + "/lib").c_str(), 1);
+  }
+  public: void LoadWorld(const std::string &_path, bool _useLevels = false)
+  {
+    this->serverConfig.SetSdfFile(
+        common::joinPaths(PROJECT_SOURCE_PATH, _path));
+    this->serverConfig.SetUseLevels(_useLevels);
+
+    this->server = std::make_unique<Server>(this->serverConfig);
+    EXPECT_FALSE(this->server->Running());
+    EXPECT_FALSE(*this->server->Running(0));
+    using namespace std::chrono_literals;
+    this->server->SetUpdatePeriod(1ns);
+  }
+
+  public: ServerConfig serverConfig;
+  public: std::unique_ptr<Server> server;
+};
+
+/////////////////////////////////////////////////
+// Load an SDF with a particle emitter and verify its properties.
+TEST_F(ParticleEmitterTest, SDFLoad)
+{
+  bool updateCustomChecked{false};
+  bool updateDefaultChecked{false};
+
+  this->LoadWorld("test/worlds/particle_emitter.sdf");
+
+  // Create a system that checks a particle emitter.
+  test::Relay testSystem;
+  testSystem.OnPostUpdate([&](const gazebo::UpdateInfo &,
+                              const gazebo::EntityComponentManager &_ecm)
+      {
+        _ecm.Each<components::ParticleEmitter,
+                  components::Name,
+                  components::Pose>(
+            [&](const ignition::gazebo::Entity &_entity,
+                const components::ParticleEmitter *_emitter,
+                const components::Name *_name,
+                const components::Pose *_pose) -> bool
+            {
+
+              if (_name->Data() == "smoke_emitter")
+              {
+                updateCustomChecked = true;
+
+                EXPECT_EQ("smoke_emitter", _name->Data());
+                EXPECT_EQ(_name->Data(), _emitter->Data().name());
+                EXPECT_EQ(msgs::ParticleEmitter_EmitterType_BOX,
+                    _emitter->Data().type());
+                EXPECT_EQ(math::Pose3d(0, 1, 0, 0, 0, 0), _pose->Data());
+                EXPECT_EQ(_pose->Data(),
+                    msgs::Convert(_emitter->Data().pose()));
+                EXPECT_EQ(math::Vector3d(2, 2, 2),
+                    msgs::Convert(_emitter->Data().size()));
+                EXPECT_DOUBLE_EQ(5.0, _emitter->Data().rate());
+                EXPECT_DOUBLE_EQ(1.0, _emitter->Data().duration());
+                EXPECT_TRUE(_emitter->Data().emitting());
+                EXPECT_EQ(math::Vector3d(3, 3, 3),
+                    msgs::Convert(_emitter->Data().particle_size()));
+                EXPECT_DOUBLE_EQ(2.0, _emitter->Data().lifetime());
+                // TODO(anyone) add material check here
```

Comment:
I wasn't sure how to load/test for materials, so I left this check out. If anyone wants to add this check in before merging, that'd be great, but if not, we can do this later.

Reply:
Leaving it for later.